app/assets/javascripts/components/Nav.js

```diff
@@ -0,0 +1,283 @@
+    if (self.viewportWidth < self.smallViewport) {
+      e.preventDefault();
+      self._toggleMobileLevel2(this);
+    }
+  });
+
+  this.$navLevel_2_Heading.click(function(e) {
+    e.preventDefault();
+    self._returnMobileNav(this);
+  });
+
+  this.$mobileNavClose.click(function(e){
+    e.preventDefault();
+    self._toggleMobileNav();
+  });
+
+  this.$mobileNavOverlay.click(function(e){
+    e.preventDefault();
+    self._toggleMobileNav();
+  });
+
+  this.$navLevel_2_Extended_Heading.click(function(e) {
+    e.preventDefault();
+    self._openMobileLevel3(this);
+  });
+
+  this.$navLevel_3_Heading.click(function(e) {
+    e.preventDefault();
+    self._closeMobileLevel3(this);
+  })
+};
+
+Nav.prototype._toggleMobileNav = function() {
+  this.$nav.toggleClass('is-active');
+  this.$mobileNavOverlay.toggleClass('is-active');
+  $('body').addClass('no-scroll');
+
+  // If we just closed the nav, reset all subnavs
+  if (!this.$nav.hasClass('is-active')) {
+    this.$nav.removeClass('is-active');
+    this.$mobileNavOverlay.removeClass('is-active');
+    this.$navLevel_1.removeClass('is-active');
+    this.$navLevel_1_item.removeClass('is-active');
+    $('body').removeClass('no-scroll');
+  }
+};
+
+Nav.prototype._toggleMobileLevel2 = function(index) {
+  var siblingsNav = $(index).siblings().get(0);
+
+  $(index)
+    .parents('[data-nav-level-1-item]').toggleClass('is-active')
+    .parents('[data-nav-level-1]').toggleClass('is-active');
+
+  $(siblingsNav).removeClass('is-hidden');
+};
+
+Nav.prototype._openMobileLevel3 = function(index) {
+  this.$navLevel_1.toggleClass('is-open');
+  $(index).siblings('[data-nav-level-3]').toggleClass('is-active');
+};
+
+Nav.prototype._closeMobileLevel3 = function(index) {
+  this.$navLevel_1.toggleClass('is-open');
+  $(index).parents('[data-nav-level-3]').toggleClass('is-active');
+};
+
+Nav.prototype._returnMobileNav = function(index) {
+  $(index)
+    .parents('[data-nav-level-1-item]').toggleClass('is-active')
+    .parents('[data-nav-level-1]').toggleClass('is-active');
+    setTimeout(function(){
+      $('[data-nav-level-2]').addClass('is-hidden');
+    }, 400);
+};
+
+/**
+* Set up events for desktop nav
+*/
+Nav.prototype._setUpDesktopInteraction = function() {
+  var self = this;
+
+  // touch event outside of global nav triggers close subnav
+  $(document).on('touchstart', function(e) {
+    if (self.viewportWidth >= self.smallViewport) {
+      if ($(e.target).parents(self.$nav).length == 0) {
+        self.$navLevel_1_item.removeClass('is-active');
+      }
+    }
+  });
+
+  this.$searchBar
+    .mouseenter(function(e) {
+      if (self.viewportWidth >= self.smallViewport) {
+        window.clearTimeout(self.timeout);
+
+        self.timeout = window.setTimeout(function() {
+          self._closeDesktopLevel2();
+        }, self.delay);
+      }
+    });
+
+  this.$navLevel_1_Heading
+    .mouseenter(function(e) {
+      if (self.viewportWidth >= self.smallViewport) {
```

Comment:
We're using this conditional a lot - can we possibly pull it out as something more readable? like `this._onDesktop()` or `this._onMobile()`

Reply:
Yes, when we implement the Dough mediaQueries methods that will be taken care of.

Reply:
Actually, have added this now.